sdk/tests/utests/Job_tests.cpp

```diff
@@ -0,0 +1,279 @@
+/**
+ * Copyright (c) 2022 Robert Bosch GmbH
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <condition_variable>
+#include <mutex>
+#include <thread>
+
+#include <gtest/gtest.h>
+
+#include "sdk/Job.h"
+
+using namespace velocitas;
+using namespace std::chrono_literals;
+
+TEST(Test_Job, callCtor_none_noFailure) {
+    EXPECT_NO_THROW(Job([] {}));
+}
+
+TEST(Test_Job, create_firstCall_noFailureAndNoNullPtr) {
+    JobPtr_t job;
+    EXPECT_NO_THROW(job = Job::create([] {}));
+    EXPECT_TRUE(job);
+}
+
+TEST(Test_Job, create_furtherCall_noFailureAndReturnsNewValidPtr) {
+    JobPtr_t job1 = Job::create([] {});
+
+    JobPtr_t job2;
+    EXPECT_NO_THROW(job2 = Job::create([] {}));
+    EXPECT_TRUE(job2);
+    EXPECT_NE(job1.get(), job2.get());
+}
+
+TEST(Test_Job, shallRecur_none_returnsFalse) {
+    Job job([] {});
+    EXPECT_FALSE(job.shallRecur());
+}
+
+TEST(Test_Job, createJob_none_passedFunctionIsNotExecuted) {
+    bool executeWasCalled = false;
+    Job  job([&executeWasCalled] { executeWasCalled = true; });
+    EXPECT_FALSE(executeWasCalled);
+}
+
+TEST(Test_Job, execute_none_passedFunctionIsExecuted) {
+    bool executeWasCalled = false;
+    Job  job([&executeWasCalled] { executeWasCalled = true; });
+
+    job.execute();
+    EXPECT_TRUE(executeWasCalled);
+}
+
+TEST(Test_Job, getFunction_notExecuted_returnedFunctionCanBeExecuted) {
```

Comment:
Well, I named it that way because the function isn't executed on the return, but actively by the test code.
Alternative would be "~returnedFunctionIsExecutable"

Reply:
As discussed: Remove this function from class Job.